design/FY2023/NFP-Space Sizing and HVAC.md

```diff
@@ -0,0 +1,265 @@
+
+## E-mail and Conference Call Conclusions ##
+
+n/a
+
+## Overview ##
+
+The `ZoneAirHeatBalanceAlgorithm` object has options for space heat balance.
+```
+ZoneAirHeatBalanceAlgorithm,
+    ThirdOrderBackwardDifference,  !- Algorithm
+    Yes,                      !- Do Space Heat Balance for Sizing
+    Yes;                      !- Do Space Heat Balance for Simulation
+```
+These currently result in additional output variables for Space, but nothing more.
+
+### Extend Spaces to Sizing
+When "Do Space Heat Balance for Sizing" is active, the following sizing changes will be active:
+
+1. Sizing results will be reported for every Space and Zone.
+2. Space sizing will follow the same method currently used for Zone sizing.
+3. Zone sizing will be based on coincident or non-coincident Space sizing results.
+4. Zone sizing results will be used to size air loops.
+5. Space sizing results will be used to size space-level HVAC if specified.
+
+
+### Extend Spaces to HVAC
+Inputs will be added to describe optional Space-level HVAC equipment, such as:
+
+1. Space air terminal units supplied by a single zone-level air distribution unit. e.g. A zone-level VAV damper supplying multiple space diffusers.
+2. Self-contained zone equipment, such as PTACs and fancoil units may be assigned to a Space. Maybe?
+3. Thermostats will still control a Zone, but they may be placed in a specific Space and will control that Space's temperature.
+4. Finish adding Space Name options to various features such as ZoneThermalChimney (if budget allows, otherwise restrict these to single-space zones).
+
+
+## Approach ##
+
+
+## Testing/Validation/Data Sources ##
+
+Compare Space vs Zone-level results.
+
+## Input Description ##
+Some new objects and some changes to existing objects are proposed.
+
+### ZoneControl:Thermostat:\*
+* *Change field "Zone or ZoneList Name" to "Zone or ZoneList or Space or SpaceList Name."*
+* When a thermostat is assigned to a zone, it will calculate the lumped loads to setpoint for all spaces in the zone.
+* When a thermostat is assigned to a space, it will calculate the loads to setpoint for that space.
+
+
+### SpaceHVAC:EquipmentConnections
+* *New object to connect nodes to a Space, similar to ZoneHVAC:EquipmentConnections.*
+
+```
+SpaceHVAC:EquipmentConnections,
+       \memo Specifies the HVAC equipment connections for a Space. Node names are specified for the
+       \memo Space air node and air inlet nodes.
+  A1 , \field Space Name
+       \required-field
+       \type object-list
+       \object-list SpaceNames
+  A2 , \field Space Air Node Name
+       \required-field
+       \type node
+  A3 ; \field Space Air Inlet Node or NodeList Name
+       \type node
+```
+
+### ZoneHVAC:SpaceSplitter
+* *New object, splits the air flow from a single piece of zone equipment to one or more Spaces.*
```

Comment:
How does the air loop splitter know how much air goes to each zone without these fraction inputs? The zone air flows create these fractions among a group of zones (i.e., it's not the splitter that does this). And if there is a splitter doesn't there need to be a mixer? My point is that I think you can use the existing splitter/mixer objects just like the air loop. The only question I have is the mechanism to get the zone (space) flow rates back to the outlet nodes of the splitter (space splitter). The answer is buried in the node maxavail data.

Reply:
Good questions. A diagram would be helpful. Currently, let's say we have a 2-zone airloop.
```
AirloopOutlet --> ZoneSplitter --> Zone1 ATU Inlet --> Zone1 ATU Outlet --> Zone 1
                              |--> Zone2 ATU Inlet --> Zone2 ATU Outlet --> Zone 2                                 
```
The ATUs control the flow and set maxavail on the ATU Inlet nodes which gets passed back up to the airloop.
The proposed method keeps all of the same nodes as above but adds the SpaceSplitter between the ATU outlet and the Space inlet nodes.
```
AirloopOutlet --> ZoneSplitter --> Zone1 ATU Inlet --> Zone1 ATU Outlet --> SpaceSplitter1--> Space 1a Supply Inlet --> Space 1a (Zone 1)
                              |                                                          |--> Space 1b Supply Inlet --> Space 1b (Zone 1)
                              |--> Zone2 ATU Inlet --> Zone2 ATU Outlet --> SpaceSplitter2--> Space 2a Supply Inlet (Zone 2)
```
So, the flow controls for the main airloop would stay the same, and the SpaceSplitter would divide up the flow to the various space diffuser outlets (which are completely passive with no controls on them). If the thermostat is located in a specific space, then the ATU will control total flow to give the controlling space the flow it needs (kind of the way unitary uses the control zone flow fraction to pro-rate the output). I'm sure there are some wholes in the scheme yet.

Reply:
In the first diagram, Zone1 and Zone 2 have design flow rates. The air loop, and the splitter, do not know those flow rates, yet the correct amount of air gets to each zone. In the second diagram, Space 1a, Space 1b, and Space 2a all have design flow rates. I don't see what's changed to need a new object. The existing splitter divides up flows without the need for flow fraction inputs. The only difference here is a ganged splitter configuration but flow is still determined from the design flow rates. Maybe in the Splitter code, space splitters are modeled first so that the zone splitters know the same information known now at the zone splitter outlet nodes. I agree you could do this either way with the same result. I think all you would need to do is place the space MassFlowRateMax and MaxAvail on the space inlet node and the existing splitter object works the same way it does now. Then model space splitters first to pass the information up to the zone splitters. Easier said than done?

Reply:
As proposed, there is no object or field to specify space design flow rates (or user specified flow rates/fractions) for the space inlets, except the SpaceSplitter. If we put a ZoneSplitter there, where would the space-level flow information reside?

Reply:
The proposed "Do Space Heat Balance for Sizing" tentatively means that given a space load you would pass a supply air T and w (from Sizing:Zone) to the space and calculate the space air flow rate.  A Tstat is needed in the space for this method or the single Tstat within the zone could be used, to get the space SPs. I'm not sure if this proposal intends to do that or not but if it does you have the space air flows and therefore the space fractions. If the space air flows are not calculated and the space loads are calculated, the space loads provide the space fractions. This seems tricky for non-coincident sizing since the space peak times may not line up with the zone peak time, and then what would happen? I think here non-coincident applies to zones, not spaces.

I assume the intent is to split the zone air flow into space flows at fixed fractions, and those flows sum to meet the coincident space peak load and air flow that match the zone. If a sizing run is done then you have the information needed to calculate the fractions. If the intent is to allow the user to "test and balance" the space diffusers (likely before they know the zone/space loads) then you would need an input. So it seems you only need these fractions when sizing is not performed. How about the AirLoopHVAC:ZoneSplitter? A single input for a space name/fraction pair list is required if the inlet node connects to a splitter and all outlet nodes connect to space inlet nodes. After doing this I'm not sure this is any better. The last 2 objects are really no better than your original suggestion except that when autosizing the FlowFractionList isn't needed. I'm also not sure what a 3rd party front end would want, use of the same object, or a specific object per use case. Proceed at will. Another thought, for non-autosized, you know the space loads prior to simulating the air systems and there is also an opportunity to calculate the space fractions. But then the space fractions would vary and that would be wierd (but all set points would be met!). Consider this discussion food for thought.

    AirLoopHVAC:ZoneSplitter,
      Zone Supply Air Splitter 1,  !- Name
      ,                        !- Space Name and Flow Fraction List
      Zone Eq In Node,         !- Inlet Node Name
      SPACE1-1 Space Splitter, !- Outlet 1 Node Name
      SPACE2-1 ATU In Node,    !- Outlet 2 Node Name
      SPACE3-1 ATU In Node,    !- Outlet 3 Node Name
      SPACE4-1 ATU In Node,    !- Outlet 4 Node Name
      SPACE5-1 ATU In Node;    !- Outlet 5 Node Name

    AirLoopHVAC:ZoneSplitter,
      Zone Space1-1 Air Splitter,  !- Name
      SPACE1-1 Flow Fraction List, !- Space Name and Flow Fraction List
      SPACE1-1 Space Splitter,     !- Inlet Node Name
      SPACE1-1a ATU In Node,       !- Outlet 1 Node Name
      SPACE1-1b ATU In Node;       !- Outlet 2 Node Name

    ZoneHVAC:FlowFractionList,
      SPACE1-1 Flow Fraction List,  !- Name
      0.4,                          !- Space Fraction for Outlet 1 Node
      0.6;                          !- Space Fraction for Outlet 2 Node

Reply:
I just looked at the Space object and a flow fraction input would work there as well.

Reply:
> I assume the intent is to split the zone air flow into space flows at fixed fractions, and those flows sum to meet the coincident space peak load and air flow that match the zone. If a sizing run is done then you have the information needed to calculate the fractions.

Yes, if autosized. 

> . If the intent is to allow the user to "test and balance" the space diffusers (likely before they know the zone/space loads) then you would need an input.

Either way, we need an autosizable input somewhere. Same as VAV terminal unit supply flow rate.